packages/typespec-vscode/src/vscode-cmd/emit-code/emit-code.ts

```diff
@@ -397,32 +397,43 @@ async function doEmit(
           ["options", emitter.package, "emitter-output-dir"],
           defaultEmitOutputDirInConfig,
         );
-      } else {
-        outputDir = emitOutputDir as string;
       }
+    }
+    const newYamlContent = configYaml.toString();
+    await writeFile(tspConfigFile, newYamlContent);
+
+    const [tspConfigOptions] = await resolveCompilerOptions(NodeSystemHost, {
+      entrypoint: mainTspFile,
+      cwd: baseDir,
+    });
+    for (const emitter of emitters) {
       let codeInfoStr: string = "code";
       if (emitter.kind !== EmitterKind.Unknown) {
         codeInfoStr = `${emitter.kind} code`;
       }
       if (emitter.language) {
         codeInfoStr += ` for ${emitter.language}`;
       }
-      outputDir = outputDir
-        .replace("{project-root}", baseDir)
-        .replace("{output-dir}", `${baseDir}/tsp-output`)
-        .replace("{emitter-name}", emitter.package);
+
+      let outputDir: string | undefined;
+      if (tspConfigOptions.options) {
+        outputDir = tspConfigOptions.options[emitter.package]["emitter-output-dir"];
+      }
       generations.push({ emitter: emitter, outputDir: outputDir, codeInfo: codeInfoStr });
     }
-    const newYamlContent = configYaml.toString();
-    await writeFile(tspConfigFile, newYamlContent);
   } catch (error: any) {
     logger.error(error);
   }
 
   const allCodesToGenerate = generations
+    .filter((g) => g.outputDir)
     .map((g) => `${g.codeInfo} under directory ${g.outputDir}`)
```

Comment:
we need to log the undefined output which is unexpected instead of filtering them.